site/src/components/PortForwardButton/PortForwardButton.tsx

```diff
@@ -0,0 +1,131 @@
+import Button from "@material-ui/core/Button"
+import Link from "@material-ui/core/Link"
+import Popover from "@material-ui/core/Popover"
+import { makeStyles } from "@material-ui/core/styles"
+import TextField from "@material-ui/core/TextField"
+import OpenInNewOutlined from "@material-ui/icons/OpenInNewOutlined"
+import { Stack } from "components/Stack/Stack"
+import { useRef, useState } from "react"
+import { colors } from "theme/colors"
+import { CodeExample } from "../CodeExample/CodeExample"
+import { HelpTooltipLink, HelpTooltipLinksGroup, HelpTooltipText } from "../Tooltips/HelpTooltip"
+
+export interface PortForwardButtonProps {
+  username: string
+  workspaceName: string
+  agentName: string
+  defaultIsOpen?: boolean
+}
+
+export const PortForwardButton: React.FC<React.PropsWithChildren<PortForwardButtonProps>> = ({
+  workspaceName,
+  agentName,
+  username,
+  defaultIsOpen = false,
+}) => {
+  const anchorRef = useRef<HTMLButtonElement>(null)
+  const [isOpen, setIsOpen] = useState(defaultIsOpen)
+  const id = isOpen ? "schedule-popover" : undefined
+  const styles = useStyles()
+  const [port, setPort] = useState("3000")
+  const { location } = window
+  const urlExample =
+    process.env.CODER_ENABLE_WILDCARD_APPS === "true"
+      ? `${location.protocol}//${port}--${workspaceName}--${agentName}--${username}.${location.host}`
+      : `${location.protocol}//${location.host}/@${username}/${workspaceName}/apps/${port}`
+
+  const onClose = () => {
+    setIsOpen(false)
+  }
```

Comment:
nit: you have a function for `onClose` but not for the the `setIsOpen` - either creating helper functions for both or just calling directly would make them more consistent.

Reply:
I just created the `onClose` because it is used in two events handlers.